src/metabase/api/user.clj

```diff
@@ -66,26 +66,91 @@
 ;;; |                   Fetching Users -- GET /api/user, GET /api/user/current, GET /api/user/:id                    |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn- status-clause
+  "Figure out what `where` clause to add to the user query when
+  we get a fiddly status and include_deactivated query.
+
+  This is to keep backwards compatibility with `include_deactivated` while adding `status."
+  [status include_deactivated]
+  (if include_deactivated
+    nil
+    (case status
+      "all" nil
+      "deactivated" [:= :is_active false]
+      "active" [:= :is_active true]
+      [:= :is_active true]
+      ))
+)
+
+(defn- query-clause
+  "Honeysql clause to shove into user query if there's a query"
+  [query]
+  [:or
+   [:like [:%lower.first_name] [(str "%" query "%")]]
+   [:like [:%lower.last_name] [(str "%" query "%")]]
+   [:like [:%lower.email] [(str "%" query "%")]]])
+
+
+(defn- user-visible-columns
+  "Columns of user table visible to current caller of API"
+  []
+  (if api/*is-superuser?* user/admin-or-self-visible-columns user/non-admin-or-self-visible-columns))
+
+(defn- user-clauses
+  "Honeysql clauses for filtering on users
+  - with a status,
+  - with a query,
+  - with a group_id,
+  - with include_deactivatved"
+  [status query group_id include_deactivated]
+  (cond-> {}
+        true (hh/merge-where (status-clause status include_deactivated))
+        true (hh/merge-where (when-let [segmented-user? (resolve 'metabase-enterprise.sandbox.api.util/segmented-user?)]
```

Comment:
This is a weird enough convention (which is legit, just unusual) that I wonder if it should become a util function (or macro, if necessary, though I doubt that).  Now appears in I think three places.

Reply:
it's always like a two-liner, tho, it seems, in all places I've found it, so is it worth the indirection, you think?

Reply:
Yeah, it's not about the number of lines so much as encapsulating what they mean.  But yeah, out of scope for this PR in any case.